otherlibs/systhreads/st_stubs.c

```diff
@@ -34,6 +34,8 @@
 #include "caml/sys.h"
 #include "caml/memprof.h"
 
+#include "../../runtime/sync_posix.h"
```

Comment:
Why is this needed?  If I comment it out it still compiles without warnings...

Reply:
Oops, `sync_posix.h` should have been included in `st_posix.h` (where it's used), not `st_pthreads.h` (where it's not).

I've pushed a correcting commit, but perhaps this is silly - `st_pthreads.h` already relies on OCaml runtime functions (i.e. it's not a pure implementation of st anymore), so an alternative would be to include `sync_posix.h` here _in `st_stubs.c`_, and not include it in any of the header files (just like the `caml/` includes) which would also remove the need for the change in `runtime/sync_posix.h`.

Reply:
I agree it's enough to include `../../runtime/sync_posix.h` once in `st_stubs.c`.  Hardening `sync_posix.h` against multiple inclusions doesn't hurt and could be useful later on.

Reply:
OK - second include deleted, but the guard in `runtime/sync_posix.h` left.